mathics/core/expression.py

```diff
@@ -367,6 +368,29 @@ def format_expr(expr):
                 return None
 
             if form != 'System`FullForm':
+                # Repeated and RepeatedNull confuse the formatter,
+                # so we need to hardlink their format rules:
+                if head == "System`Repeated":
+                    if len(leaves)==1:
+                        return Expression("System`HoldForm",
+                                      Expression("System`Postfix",
+                                      Expression(
+                                          "System`List",
+                                          leaves[0]
+                                      ),"..",170))
+                    else:
+                        return Expression("System`HoldForm",expr)
+                elif head == "System`RepeatedNull":
+                    if len(leaves)==1:
+                        return Expression("System`HoldForm",
+                                      Expression("System`Postfix",
+                                      Expression(
+                                          "System`List",
+                                          leaves[0]
+                                      ),"...",170))
+                    else:
+                        return Expression("System`HoldForm",expr)
+
```

Comment:
An interesting way to handle this. 

I suspect there is a better way, but I don't know what it is. 

And as you have said it doesn't fully address the problem of 

```
In[1] = a...
Out[1] = a...
```

But this is an improvement.